agent/structs/connect_proxy_config.go

```diff
@@ -426,6 +426,8 @@ type ExposePath struct {
 	ParsedFromCheck bool
 }
 
+// TODO: this is most likely not used. The API type is the one to be deserialized
+// with JSON
```

Comment:
Actually `agent/agent_endpoint.go` deserializes this in AgentRegisterService. Generally speaking the `api` package structs are not used by the Agent itself to deserialize incoming API json, the `agent/structs` versions are used for that, like this one.

Reply:
Ahh, that's right, thanks! I'll remove this comment.